content/lua/concepts/coroutines/coroutines.md

```diff
@@ -0,0 +1,36 @@
+---
+Title: 'Coroutines' # Required; the file name should be the same as the title, but lowercase, with dashes instead of spaces, and all punctuation removed
+Description: 'A coroutine is a line of execution, with its own stack, its own variables, and its own instruction pointer; but sharing global variables and mostly anything else with other coroutines in a collaborative manner.' # Required; ideally under 150 characters and starts with a noun (used in search engine results and content previews)
+Subjects: # Please only use Subjects in the subjects.md file (https://github.com/Codecademy/docs/blob/main/documentation/subjects.md). If that list feels insufficient, feel free to create a new Subject and add it to subjects.md in your PR!
+  - 'Computer Science'
+Tags: # Please only use Tags in the tags.md file (https://github.com/Codecademy/docs/blob/main/documentation/tags.md). If that list feels insufficient, feel free to create a new Tag and add it to tags.md in your PR!
+CatalogContent: # Please use course/path landing page slugs, rather than linking to individual content items. If listing multiple items, please put the most relevant one first
+  - 'learn-lua'
+---
+
+A **coroutine** is a line of execution, with its own stack, its own variables, and its own instruction pointer; but sharing global variables and mostly anything else with other coroutines in a collaborative manner. 
+
+
+## Understanding Coroutines
+
+A coroutine can be in one of three different states: suspended, running, or dead. This is what makes them useful, as they are essentially functions that can be paused and resumed at a later point. You can check the status of a coroutine with the status function, and resume it using the resume function. Once a coroutine is in a dead state, it cannot be resumed again, and will instead return a false and an error message when the status is checked.
+
+
+## When Should Coroutines Be Used?
+
+Coroutines are ideal when sent data needs to be matched with received data. For example, if a process needs to be run in an infinite loop thread, you can use `coroutine.yield()` to run the process, and wait for the process to resume the thread when it’s ready.
+
+
+## Functions In Coroutines
```

Comment:
@Jay01711 these would likely be better suited as terms. Thats out of scope for the internship but be sure to include that in the handoff document.

Reply:
I agree that the functions below should be deleted, and they can be added to a terms folder later. We can leave the functions header and perhaps add a sentence below it similar to:
"The following functions are used to create and manage coroutines in Lua programs:"